addons/safeStart/functions/fnc_missionTimeWarning.sqf

```diff
@@ -0,0 +1,74 @@
+/*
+ * Author: BadWolf
+ * Function used to start/stop mission length timer
+ * Sould only be called on the server
+ *
+ * Arguments:
+ * 0: Start mission length warnings <BOOL>
+ *
+ * Examples:
+ * [true] call potato_safeStart_fnc_missionTimeWarning;
+ * [false] call potato_safeStart_fnc_missionTimeWarning;
+ *
+ * Public: Yes
+ */
+
+#include "script_component.hpp"
+TRACE_1("Params",_this);
+
+if !(isServer) exitWith {};
+
+params [];
+
+missionNamespace setVariable [QGVAR(missionstartTime), CBA_missionTime, true];
+TRACE_1("Mission start time",QGVAR(missionstartTime));
+
+// 15 min Warning
+[
+    {
+        private _missionStartTime = missionNamespace getVariable QGVAR(missionstartTime);
+        private _missionLength = parseNumber getMissionConfigValue QEGVAR(missionTesting,missionTimeLength) * 60;
+        (CBA_missionTime >= (_missionStartTime + _missionLength - 900))
+    },
+    {
+        [
+            "potato_adminMsg",
+            [
+                "15 mins to mission end",
+                "Time Keeper"
+            ]
+        ] call CBA_fnc_globalEvent;
+        ["Time Keeper", "15 mins to mission end", 30] call BIS_fnc_curatorHint;
+        //End Warning
+        [
+            {
+                (CBA_missionTime >= ((_this select 0) + (_this select 1)))
+            },
+            {
+                [
+                    "potato_adminMsg",
+                    [
+                        "Mission Time has expired",
+                        "Time Keeper"
+                    ]
+                ] call CBA_fnc_globalEvent;
+            ["Time Keeper", "Mission Time has expired", 30] call BIS_fnc_curatorHint;
+            },
+            [_missionStartTime,_missionLength]
+        ] call CBA_fnc_waitUntilAndExecute;
+    },
+    []
+] call CBA_fnc_waitUntilAndExecute;
```

Comment:
it's fairly minor, but I'd rather these be `CBA_fnc_waitAndExecute` since the time delay is static
I'm happy to make the switch

Reply:
Sure thing. Can't remember if there is a specific reason that I did it this way, but if you are happy to then please do.

Reply:
Looking back there is a reason that this was used and that was in the case that SS was turned off multiple times. If we use `CBA_fnc_waitAndExcute` the warnings will be displayed at multiple times which could be confusing. I don't mind using it but it could lead to confusion as I am not aware of a way of removing the previous `waitAndExecute` in the case that SS is turned back on.

Reply:
I have changed one of these to be `waitAndExecute` and added a check in the first `waitUntilAndExcute` so that if SS gets turned on it will not duplicate the commands. I don't know of a better way to get this done.